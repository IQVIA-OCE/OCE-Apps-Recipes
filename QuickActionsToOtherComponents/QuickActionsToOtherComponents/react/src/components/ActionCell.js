import { Text } from '@oce-apps/apollo-react-native';
import React from 'react';
import { StyleSheet, TouchableOpacity, View } from 'react-native';
import { useDispatch } from 'react-redux';
import { showErrorNotification } from '../store/notification/notificationSlice';
import { openNativeViewScreen } from '../utils';

export const ActionCell = ({ row, column, ...props }) => {
  const dispatch = useDispatch();

  const openViewRecordScreen = async () => {
    try {
      await openNativeViewScreen(row.sObject, row.id);
    } catch (error) {
      dispatch(showErrorNotification(error.message));
    }
  };

  return (
    <View {...props}>
      <TouchableOpacity onPress={openViewRecordScreen}>
        <Text style={styles.text}>{row[column.accessor]}</Text>
      </TouchableOpacity>
    </View>
  );
};

const styles = StyleSheet.create({
  text: {
    color: '#0768FD',
  },
});
