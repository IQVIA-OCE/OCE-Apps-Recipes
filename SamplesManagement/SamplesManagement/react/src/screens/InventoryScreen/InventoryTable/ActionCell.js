import React, { useContext } from 'react';
import { StyleSheet, View } from 'react-native';
import { IconButton, useTheme } from 'apollo-react-native';
import { useFormikContext } from 'formik';
import { INVENTORY_FORM_TYPE } from '../../../constants/Inventories';
import { InventoryContext } from '../InventoryContext';

const ActionCell = ({ row, onPress }) => {
  const theme = useTheme();
  const { values } = useFormikContext();
  const fieldIndex = values.products.findIndex(
    ({ lotNumberId }) => row.lotNumberId === lotNumberId
  );

  const { editingType } = useContext(InventoryContext);

  return (
    <View style={styles.container}>
      {row.locked ||
      editingType === INVENTORY_FORM_TYPE.preview ||
      editingType === INVENTORY_FORM_TYPE.editSaved ? (
        <IconButton
          icon="lock"
          style={styles.icon}
          color={theme.colors.tertiary}
          size={23}
          disabled
        />
      ) : (
        <IconButton
          icon="delete"
          style={styles.icon}
          size={23}
          color={theme.colors.tertiary}
          onPress={() => onPress(row, fieldIndex)}
        />
      )}
    </View>
  );
};
const styles = StyleSheet.create({
  container: {},
  icon: { alignContent: 'center', alignItems: 'center' },
});

export default ActionCell;
