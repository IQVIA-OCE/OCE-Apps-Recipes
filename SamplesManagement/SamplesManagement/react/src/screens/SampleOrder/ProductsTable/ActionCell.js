import React from 'react';
import { StyleSheet, View } from 'react-native';
import { IconButton } from 'apollo-react-native';

const ActionCell = (props) => {
  const { row, onPress } = props;

  return (
    <View style={styles.container}>
      {row.locked ? (
        <IconButton icon="lock" style={styles.icon} size={23} disabled />
      ) : (
        <IconButton
          icon="delete"
          style={styles.icon}
          size={23}
          onPress={() => onPress(row)}
        />
      )}
    </View>
  );
};
const styles = StyleSheet.create({
  container: {},
  icon: { alignContent: 'center', alignItems: 'center' },
});

export default ActionCell;
