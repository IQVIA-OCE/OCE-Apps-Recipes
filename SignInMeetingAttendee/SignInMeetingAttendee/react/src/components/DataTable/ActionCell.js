
import React from 'react';
import { useSelector } from 'react-redux';
import { secondaryBlue, Paragraph, useTheme } from '@oce-apps/apollo-react-native';
import { View, TouchableOpacity } from 'react-native';
import { validateSignIn } from '../../utils/helper'
import { localized } from '@oce-apps/oce-apps-bridges';

const ActionCell = ({ row }) => {

    const theme = useTheme();
    const { meetingDetails, meetingGenConfig } = useSelector((state) => state.meeting);
    const { validateRestrictedFields } = row;
    const isValidateSignIn = validateSignIn(meetingGenConfig, meetingDetails, row);
    return (
        <View style={{ flexDirection: 'row' }} key={row.Id}>
            <TouchableOpacity onPress={() => validateRestrictedFields(row, true)} disabled={!isValidateSignIn} testID={'ActionCell'}>
                <View><Paragraph style={{ color: (!isValidateSignIn) ? theme.colors.text : secondaryBlue[500] }}>Sign In</Paragraph></View>
            </TouchableOpacity>
        </View>
    );
};

export default ActionCell;
