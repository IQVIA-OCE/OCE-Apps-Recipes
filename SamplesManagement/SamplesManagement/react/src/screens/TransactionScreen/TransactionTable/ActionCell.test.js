import React from 'react';
import ActionCell from './ActionCell';
import { IconButton } from '@oce-apps/apollo-react-native';
import { render, act } from '@testing-library/react-native';

describe('ActionCell', () => {
  it('should render properly', () => {
    const onPress = jest.fn();
    const Component = props => <ActionCell {...props} />;

    const { rerender, UNSAFE_root } = render(
      <Component onPress={onPress} row={{ Id: '1', locked: false }} />
    )

    act(() => UNSAFE_root.findByType(IconButton).props.onPress());

    expect(onPress).toHaveBeenCalledTimes(1);

    expect(UNSAFE_root.findByType(IconButton).props.icon).toBe('delete');

    rerender(
      <Component onPress={onPress} row={{ Id: '1', locked: true }} />
    );

    expect(UNSAFE_root.findByType(IconButton).props.icon).toBe('lock');

  });
});
