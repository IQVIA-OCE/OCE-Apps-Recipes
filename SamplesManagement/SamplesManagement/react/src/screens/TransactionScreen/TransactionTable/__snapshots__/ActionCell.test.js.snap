// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionCell should render properly 1`] = `
<View
  style={Object {}}
>
  <View
    accessibilityComponentType="button"
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessibilityTraits="button"
    accessible={true}
    centered={true}
    focusable={true}
    hitSlop={
      Object {
        "bottom": 6,
        "left": 6,
        "right": 6,
        "top": 6,
      }
    }
    onClick={[Function]}
    onLayout={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    pointerEvents="box-only"
  >
    <View
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "margin": 6,
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 17.25,
              "height": 34.5,
              "width": 34.5,
            },
            undefined,
            Object {
              "alignContent": "center",
              "alignItems": "center",
            },
          ],
        ]
      }
    >
      <View>
        <Text
          accessibilityElementsHidden={true}
          allowFontScaling={false}
          importantForAccessibility="no-hide-descendants"
          pointerEvents="none"
          style={
            Array [
              Object {
                "color": "#000000",
                "fontSize": 23,
              },
              Array [
                Object {
                  "transform": Array [
                    Object {
                      "scaleX": 1,
                    },
                  ],
                },
                Object {
                  "backgroundColor": "transparent",
                },
              ],
              Object {
                "fontFamily": "Material Design Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "bottom": 0,
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "borderRadius": 0,
          },
        ]
      }
    />
  </View>
</View>
`;

exports[`ActionCell should render properly 2`] = `
<View
  style={Object {}}
>
  <View
    accessibilityComponentType="button"
    accessibilityRole="button"
    accessibilityStates={
      Array [
        "disabled",
      ]
    }
    accessibilityTraits={
      Array [
        "button",
        "disabled",
      ]
    }
    accessible={true}
    centered={true}
    focusable={true}
    hitSlop={
      Object {
        "bottom": 6,
        "left": 6,
        "right": 6,
        "top": 6,
      }
    }
    onClick={[Function]}
    onLayout={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    pointerEvents="box-only"
  >
    <View
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "margin": 6,
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 17.25,
              "height": 34.5,
              "width": 34.5,
            },
            Object {
              "opacity": 0.32,
            },
            Object {
              "alignContent": "center",
              "alignItems": "center",
            },
          ],
        ]
      }
    >
      <View>
        <Text
          accessibilityElementsHidden={true}
          allowFontScaling={false}
          importantForAccessibility="no-hide-descendants"
          pointerEvents="none"
          style={
            Array [
              Object {
                "color": "#000000",
                "fontSize": 23,
              },
              Array [
                Object {
                  "transform": Array [
                    Object {
                      "scaleX": 1,
                    },
                  ],
                },
                Object {
                  "backgroundColor": "transparent",
                },
              ],
              Object {
                "fontFamily": "Material Design Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "bottom": 0,
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "borderRadius": 0,
          },
        ]
      }
    />
  </View>
</View>
`;
